toolchain/ltrace.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python
+
+import os
+from avocado import Test
+from avocado import main
+from avocado.utils import build
+from avocado.utils import process
+from avocado.utils import git
+from avocado.utils import distro
+from avocado.utils.software_manager import SoftwareManager
+
+backend = SoftwareManager()
+dist = distro.detect()
+
+
+class Ltrace(Test):
+
+    def setUp(self):
+        """
+        Install  all the required dependencies
+        Building source tarball requires packages specific to os
+        that needs to be installed, if not installed test will stop
+        a library function check_repo() should be implemented.
+        """
+
+        os_deps = ['git', 'wget', 'rpm', 'autoconf',
+                   'automake', 'dejagnu', 'binutils']
+
+        if dist.name == 'sles':
+            sles_deps = ['build', 'libdw-devel', 'libelf-devel',
+                         'elfutils', 'binutils-devel', 'libtool', 'gcc-c++']
+            pkgs = os_deps + sles_deps
+
+        elif dist.name == 'redhat':
+            rhel_deps = ['elfutils-devel', 'elfutils-libelf-devel',
+                         'elfutils-libelf', 'elfutils-libs', 'libtool-ltdl']
+            pkgs = os_deps + rhel_deps
+
+        elif dist.name == 'ubuntu':
+            ubuntu_deps = ['elfutils', 'libelf-dev', 'libtool',
+                           'libelf1', 'librpmbuild3', 'binutils-dev']
+            pkgs = os_deps + ubuntu_deps
+        else:
+            print "Unsupported OS!"
+
+        i = 0
+        while i < len(pkgs):
+            if backend.install(pkgs[i]):
+                print "%s installed successfull" % pkgs[i]
+            else:
+                print "Error: Check repository !"
+                exit(1)
+            i = i + 1
+
+        """
+        clone the  ltrace tests from the GIT repo
+        Source: git clone git://git.debian.org/git/collab-maint/ltrace.git
+        """
+
+        git.get_repo('git://git.debian.org/git/collab-maint/ltrace.git',
+                     destination_dir=self.srcdir + '/ltrace')
+        ltrace_version = os.path.basename("ltrace")
+        os.chdir(self.srcdir)
+
+        self.srcdir = os.path.join(self.srcdir, ltrace_version)
+        os.chdir(self.srcdir)
+        process.run('./autogen.sh')
+        process.run('./configure')
+        build.make(self.srcdir)
+        try:
+            process.run('which gcc')
+        except error.CmdError:
+            raise NotAvailableError('gcc is required by this job and is '
+                                    'not available on the system')
+
+    def test(self):
+        os.chdir(self.srcdir)
+        build.make(self.srcdir, extra_args='check')
```

Comment:
Basically the output of this test is similar to binutils, so let me paste the suggestion here:

This runs the checks and produces huge output, but one does not see the details. So basically this test gives you the PASS/FAIL, but later on you can't tell why it failed. Would you please copy the `*.sum` files from the `binutils` build directory to the test results to allow postprocess?

Additionally it would be convenient to list the failed cases if it fails. Manually I do `for summary in $(find . -name '*.sum'); do echo $summary; grep ^FAIL $summary; done` in the build directory.

So I'd suggest:

```python
for root, dirnames, filenames in os.walk('.'):
    for filename in fnmatch.filter(filenames, '*.sum'):
        filename = os.path.join(root, filename)
        logfile = filename[:-4] + ".log"
        # copy logfile to self.logdir
        # open(filename); read line-by-line and self.log.error lines starting with `FAIL`
        # (alternatively attach the full filename as it might be useful to see what tests were skipped)
```

What do you think? (I'm not enforcing you to do that, I just think it'd be useful to attach those files).